### PR TITLE
Enables perssist-message flag in the store protocol for wakunode2

### PR DESCRIPTION
**Pull request by staheri14**
Addresses https://github.com/status-im/nim-waku/issues/494
It introduces `persist-messages` flag to activate the store protocol  in the light mode. It defaults to `false`.
This flag impacts wakunode2 and disables persisting messages in a database. 

However, as far as I know, in chat2  there is no option to persist messages. Please correct me if I am mistaken @jm-clius @oskarth .
In chat2, the store protocol is mounted in https://github.com/status-im/nim-waku/blob/7e9dac67c8927c953687ee6391c63286377bb9c9/examples/v2/chat2.nim#L280
which invokes this proc whose `store` field is Nil by default
https://github.com/status-im/nim-waku/blob/7e9dac67c8927c953687ee6391c63286377bb9c9/waku/v2/node/wakunode2.nim#L362

### CODE REVIEW

**Review by oskarth — APPROVED** [review]
Probably want to set `store` to true as well by default

**Comment by jm-clius** [comment]
> in chat2 there is no option to persist messages
Correct. `chat2` should retrieve history from one of the cluster nodes

**Review by jm-clius — COMMENTED** [review]
Though this allows disabling persisting messages to the database, a node with `store=true` will still locally "store" messages in-memory in the `messages` sequence. I understood from [the previous discussion](https://github.com/status-im/nim-waku/issues/494#issuecomment-825706738) that the flag would cover this case as well, or do we want to configure this separately?

**Comment by oskarth** [comment]
Correct, that case should ideally be covered as well

**Comment by staheri14** [comment]
> Though this allows disabling persisting messages to the database, a node with `store=true` will still locally "store" messages in-memory in the `messages` sequence. I understood from [the previous discussion](https://github.com/status-im/nim-waku/issues/494#issuecomment-825706738) that the flag would cover this case as well, or do we want to configure this separately?

Thanks @jm-clius and @oskarth ! 

Further changes on the PR to be reviewed:

1. The functionality of the light node is now limited to sending history query and getting response. 
The storage of waku messages is totally disabled. This is done by disabling the subscription part when the persist-messages flag is false. This means a light node will not receive any waku messages hence wont store any messages. (Alternatively, instead of disabling the subscription,  an empty handler can be set when light mode is active . However, I am not sure about any future usecase for such implementation. wdyt?)

2. The default of `persist-messages` is also set to `true`, this is because persistent storage of messages has been a default behaviour of the store protocol and is used in this way in all the waku v2 tutorials and instructions. Light nodes will have to set the `persist-messages` to false.

3- A similar  `persist-messages` flag is included in the config_bridge, to be able to configure the mode of store protocol while bridging between v1 and v2, please let me know whether the use of flag in this context is correct, or should the store protocol be always mounted in the full mode? @jm-clius

File: waku/v2/node/config.nim [hunk 1]
```diff
@@ -55,12 +55,12 @@ type
 
     persistmessages* {.
       desc: "Enable message persistence: true|false",
-      defaultValue: false
+      defaultValue: true
```
Comment (oskarth) [inline waku/v2/node/config.nim:58]: Do we want this to be true? To me the most common path is to be a light store client, but I guess one could argue both ways

**Review by oskarth — APPROVED** [review]
> The default of persist-messages is also set to true, this is because persistent storage of messages has been a default behaviour of the store protocol and is used in this way in all the waku v2 tutorials and instructions. Light nodes will have to set the persist-messages to false.

Intuitively I'd probably set this to false, especially since `store` was false by default before. For the tutorial, I'd just update the `store.md` one to replace that flag with `persist-mesages`. WDYT? It isn't a big deal though and we could also change this later if we want to.

File: waku/v2/node/config.nim (hunk 1 above)
Comment (jm-clius) [reply waku/v2/node/config.nim:58]: I'm leaning towards `false` as well. In my view, the most common "default" client is one that starts up, retrieves history as a "light" store node and participates in the network as a relay node. Thus, defaults of `store=true`, `relay=true`, `persist-messages=false`. Such clients will, of course, have to be configured with a `storenode` as well.

**Review by jm-clius — APPROVED** [review]
LGTM! There is just the conversation around `persist-messages` being `false` default. As for
> similar persist-messages flag is included in the config_bridge

<-- absolutely fine as is. I don't foresee a bridge ever doubling as a storenode (which is why `store` should remain `false` by default for the bridge), but it's good to keep the config options up to date

**Comment by staheri14** [comment]
Thanks @oskarth and @jm-clius 
I updated the defaults to `store=true`, `persist-messages=false` both in wakunode2 (and chat2 which inherits the wakunode2 config) and in the bridge config. 
The instructions around the `persist-messages` is also included in the `store.md` file as well as `chat2.md`.

I would be thankful if you please have a quick look at these recent  updates and feel free to merge this PR if things look OK.

**Review by oskarth — APPROVED** [review]

**Review by jm-clius — APPROVED** [review]

**Pull request merged**